test/compilable/ddoc_markdown_code.d

```diff
@@ -0,0 +1,45 @@
+// PERMUTE_ARGS:
+// REQUIRED_ARGS: -D -Dd${RESULTS_DIR}/compilable -o- -preview=markdown
+// POST_SCRIPT: compilable/extra-files/ddocAny-postscript.sh
+
+/++
+# Code Blocks
+
+---
+A regular ol' code block (note the uneven delimiter lengths)
+----
+
+```
+A backtick-fenced code block
```

Comment:
These blocks have usually no syntax highlighting.

Reply:
This PR leaves backtick-delimited code sections like `` `variableName` `` working the same as they currently do with no syntax highlighting, but also adds the ability to add a backtick-fenced (or tilde-fenced) code block that behaves the same as the existing hyphen-fenced code block. So unless it specifies a different language, the backtick-fenced code block would gain D syntax highlighting.

Reply:
Yeah, I understood that. But that's usually not how Markdown works. If no language is specified now syntax highlighting is done.

Reply:
Ah, sorry for misreading you earlier. It's true that standard Markdown doesn't usually work that way. However because this is part of documentation for the D language, to me it makes sense that 1) fenced code blocks without a language specifier should be assumed to be D and that 2) code blocks with different kinds of fence delimiters should behave the same as each other. Since hyphen-delimited code blocks already do D syntax highlighting, I think the others should follow suit.

Reply:
If I don't want syntax highlighting then I would need to specify a language, even if it's not correct. That doesn't feel right.

Reply:
And yet that's the current behavior with fenced code blocks, with the added inability to opt out at all.

I know we could make the new fences behave differently than the hyphen fences, but I don't like adding special cases for developers to recall. And I suspect that in most documentation the frequency that you want to avoid syntax highlighting is far less than the frequency you want D highlighting.

To avoid syntax highlighting the following doesn't look bad:

```
--- don't syntax highlight
Don't highlight this code!
---
```

Or of course you could just use a macro.

Reply:
Code blocks with three dashes are not markdown, as far as I know. The CommonMark spec says:

"A code fence is a sequence of at least three consecutive backtick characters (`) or tildes (~)"

Reply:
Certainly--consecutive dashes are used for horizontal rules and for setext headings in regular Markdown. But this isn't a full Markdown parser—this is adding some Markdown features to Ddoc, and where they overlap existing features my inclination is to make them match as well as possible. My opinion is that I don't like making everyone need to recall that one kind of code fence highlights by default and another kind doesn't, unless there's a compelling reason for the difference.

Reply:
The reason would be to follow Markdown as close as possible. I have no other reason than that.

Reply:
I'm with Jacob on this btw, the three ticks should NOT be syntax highlighted. If you want D syntax highlighting, use the D three dashes.